src/test/java/com/teragrep/aer_01/EventContextConsumerTest.java

```diff
@@ -0,0 +1,145 @@
+                eventContext = eventContextFactory.create();
+                eventContextConsumer.accept(eventContext);
+            }
+
+            long latency = Instant.now().getEpochSecond();
+
+            // 5 records for each partition
+            Gauge<Long> gauge1 = metricRegistry.gauge(name(EventContextConsumer.class, "latency-seconds", "1"));
+            Gauge<Long> gauge2 = metricRegistry.gauge(name(EventContextConsumer.class, "latency-seconds", "2"));
+
+            // hard to test the exact correct latency
+            Assertions.assertTrue(gauge1.getValue() >= latency);
+            Assertions.assertTrue(gauge2.getValue() >= latency);
+        }
+    }
+
+    @Test
+    public void testDepthBytesMetric() throws Exception {
+        final EventContextFactory eventContextFactory = new CheckpointlessEventContextFactory();
+        final Sourceable configSource = new PropertySource();
+        final int prometheusPort = new MetricsConfig(configSource).prometheusPort;
+        final MetricRegistry metricRegistry = new MetricRegistry();
+
+        try (EventContextConsumer eventContextConsumer = new EventContextConsumer(configSource, new Output.FakeOutput(), metricRegistry, prometheusPort)) {
+            EventContext eventContext = eventContextFactory.create();
+            eventContextConsumer.accept(eventContext);
+
+            long depth1 = 0L;
+            final double records = 10;
+            for (int i = 0; i < records - 1; i++) {
+                eventContext = eventContextFactory.create();
+                eventContextConsumer.accept(eventContext);
+
+                if (i == 4) {
+                    depth1 = eventContext.getLastEnqueuedEventProperties().getOffset() - eventContext.getEventData().getOffset();
+                }
+            }
+
+            long depth2 = eventContext.getLastEnqueuedEventProperties().getOffset() - eventContext.getEventData().getOffset();
+            Gauge<Long> gauge1 = metricRegistry.gauge(name(EventContextConsumer.class, "depth-bytes", "1"));
+            Gauge<Long> gauge2 = metricRegistry.gauge(name(EventContextConsumer.class, "depth-bytes", "2"));
+
+            Assertions.assertEquals(depth1, gauge1.getValue());
+            Assertions.assertEquals(depth2, gauge2.getValue());
+        }
+    }
+
+    @Test
+    public void testEstimatedDataDepthMetric() throws Exception {
+        final EventContextFactory eventContextFactory = new CheckpointlessEventContextFactory();
+        final Sourceable configSource = new PropertySource();
+        final int prometheusPort = new MetricsConfig(configSource).prometheusPort;
+        final MetricRegistry metricRegistry = new MetricRegistry();
+
+        try (EventContextConsumer eventContextConsumer = new EventContextConsumer(configSource, new Output.FakeOutput(), metricRegistry, prometheusPort)) {
+            final double records = 10;
+            long length = 0L;
+            for (int i = 0; i < records; i++) {
```

Comment:
This seems to be working as expected but change to follow the suit if the others fors are changed